arch/risc-v/src/common/supervisor/riscv_vectors.S

```diff
@@ -0,0 +1,52 @@
+/****************************************************************************
+ * arch/risc-v/src/common/supervisor/riscv_vectors.S
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+  .section .text
+  .balign  8
+  .global  __trap_vec
+  .global  __irq_vec
+
+/****************************************************************************
+ * Name: __trap_vec
+ *
+ * Description:
+ *   All M-mode exceptions and interrupts will be handled from here.
+ *
+ ****************************************************************************/
+
+__trap_vec:
```

Comment:
could be shared with the machine version?

Reply:
In this case this is the wrong place to declare trap_vec anyway.

The implementation of the machine level trap handler is different depending on if the NuttX kernel runs in M-mode directly. 

Compared to if it is S-mode, which requires some kind of SBI (which will take ownership of the trap handler).

So I will remove this from here anyway, as it does not belong here

Reply:
Done